pyvista/plotting/widgets.py

```diff
@@ -705,17 +729,19 @@ def add_mesh_slice(
             VTK actor of the mesh.
 
         """
+        mesh, algo = algorithm_to_mesh_handler(mesh)
+
         name = kwargs.get('name', mesh.memory_address)
         rng = mesh.get_data_range(kwargs.get('scalars', None))
         kwargs.setdefault('clim', kwargs.pop('rng', rng))
         mesh.set_active_scalars(kwargs.get('scalars', mesh.active_scalars_name))
         if origin is None:
             origin = mesh.center
 
-        self.add_mesh(mesh.outline(), name=f"{name}-outline", opacity=0.0)
+        self.add_mesh(outline_algorithm(algo or mesh), name=f"{name}-outline", opacity=0.0)
```

Comment:
Well when I have `algo or mesh`, I am relying on the `or` logic so that if we are using an algorithm it is passed but otherwise falls back to the `mesh`

Reply:
I just meant pre-defining a name for `algo or mesh` so that the cognitive burden of resolving this logic is moved to one place in the method. But this is probably not worth it (and probably refactored soon anyway). Let's leave it.